homeassistant/helpers/selector.py

```diff
@@ -1219,17 +1220,22 @@ class TextSelector(Selector[TextSelectorConfig]):
                 vol.Coerce(TextSelectorType), lambda val: val.value
             ),
             vol.Optional("autocomplete"): str,
+            vol.Optional("multiple", default=False): bool,
         }
     )
 
     def __init__(self, config: TextSelectorConfig | None = None) -> None:
         """Instantiate a selector."""
         super().__init__(config)
 
-    def __call__(self, data: Any) -> str:
+    def __call__(self, data: Any) -> str | list[str]:
         """Validate the passed selection."""
-        text: str = vol.Schema(str)(data)
-        return text
+        if not self.config["multiple"]:
+            text: str = vol.Schema(str)(data)
+            return text
+        if not isinstance(data, list):
+            raise vol.Invalid("Value should be a list")
```

Comment:
Should we make it a list instead at that point? (e.g., ensure a list?)

Reply:
area, entity, device and select selectors raise an error so I used the same logic. Do you think we need to change for all these selectors?

Reply:
Nah it is fine then 👍